core/arch/arm/mm/tee_mmu.c

```diff
@@ -722,6 +723,7 @@ TEE_Result tee_mmu_user_pa2va_helper(const struct user_ta_ctx *utc,
 
 		if (core_is_buffer_inside(pa, 1, p,
 					  utc->mmu->regions[n].size)) {
+			p = ROUNDDOWN(p, SMALL_PAGE_SIZE);
 			*va = (void *)(pa - p + utc->mmu->regions[n].va);
```

Comment:
I don't see how this function can work at all with physically noncontiguous mobjs. Since this function is unused in practice perhaps we should just return an error instead.

Reply:
This is more tricky than expected. We must parse all possible small page phys block to check if one matches the target `pa` through `core_is_buffer_inside`.